neqo-http3/src/connection_client.rs

```diff
@@ -299,14 +327,46 @@ impl Http3Client {
                 Ok((amount, fin))
             }
             Err(e) => {
-                if e == Error::HttpFrame {
+                if e.connection_error() {
                     self.close(now, e.code(), "");
                 }
                 Err(e)
             }
         }
     }
 
+    // API: Push streams
+
+    /// Cancel a push
+    /// # Errors
+    /// `InvalidStreamId` if the stream does not exist.
+    pub fn cancel_push(&mut self, push_id: u64) -> Res<()> {
+        self.push_handler
+            .borrow_mut()
+            .cancel(push_id, &mut self.conn, &mut self.base_handler)
+    }
+
+    /// Push response data are read directly into a buffer supplied as a parameter of this function
+    /// to avoid copying data.
+    /// # Errors
+    /// It returns an error if a stream does not exist(`InvalidStreamId`) or an error has happened while
+    /// reading a stream, e.g. early close, protocol error, etc.
+    pub fn push_read_data(
+        &mut self,
+        now: Instant,
+        push_id: u64,
```

Comment:
I think that we want a newtype for PushId, just like StreamId.

Reply:
Let's do that in a separate PR, this one is too big and I really want to land it.
I have open issue #785